lib/aws-record/record/item_operations.rb

```diff
@@ -174,11 +210,18 @@ def update!(new_params, opts = {})
       # instance in Amazon DynamoDB. Uses the
       # {http://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/DynamoDB/Client.html#delete_item-instance_method
       # Aws::DynamoDB::Client#delete_item} API.
-      def delete!
-        dynamodb_client.delete_item(
+      #
+      # @param [Hash] opts
+      # @option opts [Hash] :delete_item_options Options to pass to the
+      # {http://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/DynamoDB/Client.html#delete_item-instance_method
+      # Aws::DynamoDB::Client#delete_item} call.
+      def delete!(opts = {})
+        delete_item_options = opts.delete(:delete_item_options) || {}
+        delete_opts = {
           table_name: self.class.table_name,
           key: key_values
-        )
+        }.merge(delete_item_options)
```

Comment:
This does let you pass in a table_name or a key and override the values that should be getting used here.  That seems potentially risky / error prone.  Possibly we should switch the order of the merge (so that the values we compute here should always be set that).

Reply:
Yeah I wanted it to be a final override. It's more complex in other cases. Should we have it work like defaults, where these values get overridden instead?

Reply:
Changed it to work as a default.